lib/client/local_proxy_middleware.go

```diff
@@ -119,48 +165,18 @@ type CertIssuer interface {
 
 // DBCertIssuer checks and issues db certs.
 type DBCertIssuer struct {
-	// Client is a TeleportClient used to issue certificates when necessary.
-	Client *TeleportClient
-	// RouteToApp contains database routing information.
-	RouteToApp tlsca.RouteToDatabase
+	IssueCertFunc func(ctx context.Context) (tls.Certificate, error)
```

Comment:
nit: I'm not a fan of adding `IssueCertFunc` here. What if we kept `DBCertIssuer` as is and introduced a new implementation for gateways - `GatewayCertIssuer`? This would also help reduce conflicts with my open PR - https://github.com/gravitational/teleport/pull/40985

Reply:
I wasn't sure if extracting `IssueCertFunc` is a correct direction, thanks for the feedback.

OTOH, now I'm not sure if `GatewayCertIssuer` would be any better than the old `dbMiddleware`. In the end Connect and tsh would have separate implementations for the middleware, no? They would only live in the same module.

Reply:
Hmm I think there is some value in at least utilizing the `CertChecker` type with a new `CertIssuer` just for the middleware piece, but holding off for now would be fine too. I'd rather we wait to introduce this properly, whether it's converting`CertChecker` into a generic type or otherwise cleanly abstracting the cert issuance logic.

Reply:
Okay, I think I will revert this refactoring. It's not necessary for per-session MFA and we can always come back to it in the future.